tokio-quiche/src/http3/settings.rs

```diff
@@ -0,0 +1,167 @@
+use std::future::poll_fn;
+use std::task::{Context, Poll};
+use std::time::Duration;
+
+use crate::http3::driver::H3ConnectionError;
+use crate::quic::QuicheConnection;
+
+use foundations::telemetry::log;
+use tokio_util::time::delay_queue::{self, DelayQueue};
+
+/// Unified configuration parameters for [H3Driver](crate::http3::driver::H3Driver)s.
+#[derive(Default, Clone, Debug)]
+pub struct Http3Settings {
+    /// Maximum number of requests a [ServerH3Driver](crate::http3::driver::ServerH3Driver)
+    /// allows per connection.
+    pub max_requests_per_connection: Option<u64>,
+    /// Maximum size of a single HEADERS frame, in bytes.
+    pub max_header_list_size: Option<u64>,
```

Comment:
The name and description are not aligned with RFC 9114. However, we might want to punt on resolving this.